### ISSUE TITLE
Assess continued use of TorchData

### ISSUE THREAD

**Issue by atolopko-czi** [chanzuckerberg/cellxgene-census#685]
The TorchData library that the Census ExperimentDataPipe is implemnted upon is no longer being actively developed: https://github.com/pytorch/data/issues/1196.

Note that there are two separate `IterDataPipe` classes:
* `torch.utils.data.datapipes.datapipe.IterDataPipe`
* `torchdata.datapipes.iter.IterDataPipe`

Census API's `ExperimentDataPipe` uses the latter, currently. Even if we were to simply switch to using`torch.utils.data.datapipes.datapipe.IterDataPipe`, the larger issue is the expectation that TorchData DataPipe "operations" like `RandomSplitter` and  `Shuffler` are available for use in client code. So we may want to provide equivalent, custom functionality in our `ExperimentDataPipe`, which would be unfortunate.

We can also keep our implementation "as is", since TorchData isn't being pulled away, it just won't see any improvements.

We certainly need to keep `ExperimentDataPipe` as an "experimental" Census API until we resolve this issue.

**Comment by mmdanziger** [comment]
Please note that as of release [0.10.1]
(https://github.com/pytorch/data/releases/tag/v0.10.1) torchdata has deleted the datapipes
cf https://github.com/pytorch/data/issues/1196

Currently cellxgene census does not work on a clean install

**Comment by atolopko-czi** [comment]
@mmdanziger Thank you for reporting the breakage. The PyTorch support within this repo is in need of deprecation and removal. PyTorch support for SOMA data objects (the data specification upon which CELLxGENE Census is built) is being migrated to [TileDB-SOMA-ML](https://github.com/single-cell-data/TileDB-SOMA-ML). And while this is still a a pre-release package and subject to breaking API changes prior to its first release, it is being actively developed.
